File: lab_1/src/SumThread.java
public class SumThread extends Thread {
    private final int id;
    private final int step;
    volatile private boolean running = true;

    private long sum = 0;
    private long count = 0;

    public SumThread(int id, int step) {
        this.id = id;
        this.step = step;
    }

    @Override
    public void run() {
        int current = 0;
        while (running) {
            sum += current;
            current += step;
            count++;

//            try {
//                Thread.sleep(100);
//            } catch (InterruptedException e) {
//                return;
//            }
        }

        System.out.printf("Потік #%d завершено. Сума = %d, доданків = %d%n", id, sum, count);
    }

    // Метод для зупинки потоку
    public void stopRunning() {
        running = false;
    }
}

File: lab_1_C#/SumThread.cs
using System;
using System.Threading;

class SumThread
{
    private readonly int id;
    private readonly int step;
    private volatile bool running = true;

    private long sum = 0;
    private long count = 0;

    private Thread thread;

    public SumThread(int id, int step)
    {
        this.id = id;
        this.step = step;
    }

    public void Start()
    {
        thread = new Thread(Run);
        thread.Start();
    }

    private void Run()
    {
        int current = 0;

        // Запам'ятовуємо стартовий час
        long startTime = DateTimeOffset.Now.ToUnixTimeMilliseconds();

        while (running)
        {
            sum += current;
            current += step;
            count++;

            // Перевіряємо час на кожному кроці
            long currentTime = DateTimeOffset.Now.ToUnixTimeMilliseconds();
            if (currentTime - startTime >= 100) // Перевіряємо, чи пройшло 100 мс
            {
                startTime = currentTime; // Оновлюємо час
            }
        }

        Console.WriteLine($"Потiк #{id} завершено. Сума = {sum}, доданків = {count}");
    }

    public void StopRunning()
    {
        running = false;
    }

    public void Join()
    {
        thread.Join();
    }
}